proxy_manager.go

```diff
@@ -29,7 +29,7 @@ import (
 
 type proxyManager struct {
 	mu              *sync.RWMutex
-	proxies         map[string]interface{}
+	proxies         *sync.Map
```

Comment:
According to https://pkg.go.dev/sync#Map:


```
The Map type is optimized for two common use cases: (1) when the entry for a given key is only ever written once but read many times, as in caches that only grow, or (2) when multiple goroutines read, write, and overwrite entries for disjoint sets of keys. In these two cases, use of a Map may significantly reduce lock contention compared to a Go map paired with a separate Mutex or RWMutex.

```

Is our case one of these two cases? If not, maybe we should not change map + RWMutex usage to sync.Map. Can you explain the reasoning of the change more in detail ? I see some information in the jira ticket. It may make sense to copy it to the PR desc. 


Looking at the results in the jira ticket, I don't see any improvement. It seems like latencies stayed the same.

I did not get the `operations with t >= 0.5ms` benchmark as well. What is an operation here?

Reply:
I've updated the Jira ticket.

Readers block when a writer holds the RWMutex, that increases latencies in this particular scenario where `GetMap` is called with thousands of names concurrently. `sync.Map` has better latencies with that.

Since the map used for local proxy cache grows but the items there are not updated, `sync.Map` performs better.

Reply:
I asked something about numbers in Jira ticket, after that I am ok with this PR.